src/pages/instances/Events.tsx

```diff
@@ -65,45 +71,60 @@ const Events: FC = () => {
     return "undefined";
   };
 
-  const connectEventWs = () => {
-    const wsUrl = `wss://${location.host}/1.0/events?type=operation,lifecycle&all-projects=true`;
-    const ws = new WebSocket(wsUrl);
-    ws.onopen = () => {
-      setEventWs(ws);
-    };
-    ws.onclose = () => {
-      setEventWs(null);
-    };
-    ws.onmessage = (message: MessageEvent<Blob | string | null>) => {
-      if (typeof message.data !== "string") {
-        console.log("Invalid format on event api: ", message.data);
-        return;
-      }
-      const event = JSON.parse(message.data) as LxdEvent;
-      if (event.type === "operation") {
-        queryClient.invalidateQueries({
-          queryKey: [queryKeys.operations, event.project],
-        });
-        refetchOperations();
-      }
-      if (event.type === "lifecycle") {
-        const rootQueryKey = getLifecycleRootQueryKey(event);
-        queryClient.invalidateQueries({
-          predicate: (query) => query.queryKey[0] === rootQueryKey,
-        });
+  const connectEventWs = (retryCount = 0) => {
+    try {
+      const wsUrl = `wss://${location.host}/1.0/events?type=operation,lifecycle&all-projects=true`;
+      const ws = new WebSocket(wsUrl);
+      ws.onopen = () => {
+        setEventWs(ws);
+      };
+      ws.onclose = () => {
+        setEventWs(null);
+      };
+      ws.onmessage = (message: MessageEvent<Blob | string | null>) => {
+        if (typeof message.data !== "string") {
+          console.log("Invalid format on event api: ", message.data);
+          return;
+        }
+        const event = JSON.parse(message.data) as LxdEvent;
+        if (event.type === "operation") {
+          queryClient.invalidateQueries({
+            queryKey: [queryKeys.operations, event.project],
+          });
+          refetchOperations();
+        }
+        if (event.type === "lifecycle") {
+          const rootQueryKey = getLifecycleRootQueryKey(event);
+          queryClient.invalidateQueries({
+            predicate: (query) => query.queryKey[0] === rootQueryKey,
+          });
+        }
+        // ensure open requests that reply with an operation and register
+        // new handlers in the eventQueue are closed before handling the event
+        setTimeout(() => {
+          handleEvent(event);
+        }, EVENT_HANDLER_DELAY);
+      };
+    } catch (e) {
+      if (retryCount < MAX_WS_CONNECTION_RETRIES) {
```

Comment:
Need to ensure this won't delay surfacing connection error for users with restricted permissions.

Also the need for reconnect in LXD is slightly lower than in anbox. Because here we only ever connect to a single endpoint. Unlike anbox connecting to multiple sub clusters where some might be unavailable.

Reply:
Ok, let me know and in case of any problem I can revert this part. But if QA works with no problems, I think having a reconnection mechanism is good even for the single endpoint :)

Reply:
All good after QA check.

Reply:
Ok, if there are no changes needed feel free to merge whenever you want - I don't have the authorisation to merge myself :)